src/edu/ucsb/cs56/projects/games/subsink/Sub.java

```diff
@@ -80,7 +109,17 @@ public void finalize(World w) {
 		w.giveScore(1);
 	}
 
+
+
 	public void paint(Graphics2D g) {
-		g.drawImage(ImageLoader.get("img/sub.png"), (int)x, (int)y, null);
+		double xOffset = (int) (Math.random() * 10) + x - 10;
+		double yOffset = (int) (Math.random() * 10) + y - 10;
+		if (isExploded) {
+			g.drawImage(ImageLoader.get("img/explosion.png"), (int) xOffset, (int) yOffset, null);
+		}else if (flipped){
```

Comment:
I explained this to durva, but there's absolutely no need for there to be a `flipped` class variable. The less duplicate state being stored the better. Replace this with `xSpeed < 0` and delete the variable.